src/tests/junit/org/apache/tools/ant/types/CharSetTest.java

```diff
@@ -0,0 +1,20 @@
+package org.apache.tools.ant.types;
+
+import org.apache.tools.ant.BuildException;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+public class CharSetTest {
+    @Test
+    public void testCorrectNames() {
+        String[] expected = {"UTF-8", "ISO-8859-1", "037", "us", "IBM500"};
+        Arrays.stream(expected).forEach(new CharSet()::setValue);
+    }
+
+    @Test(expected = BuildException.class)
+    public void testNonExistentNames() {
+        String[] nonexistent = {"mojibake", "dummy"};
+        Arrays.stream(nonexistent).forEach(new CharSet()::setValue);
```

Comment:
is the test for "dummy" ever going to be run?

Reply:
I was lazy... 😁 the tests are properly parameterised now